### PR TITLE
Move pre-reqs to operator

### PR DESCRIPTION
**Pull request by amdabhad**
## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature Enhancement
- [ ] Test Automation 
- [ ] Code Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Community Operator listing 
- [ ] Other (please describe): 

## What is the current behavior?

-

## What is the new behavior?

-

## How risky is this change?
- [ ] Small, isolated change
- [ ] Medium, requires regression testing
- [x] Large, requires functional and regression testing

### CODE REVIEW

**Comment by amdabhad** [comment]
Driver image: quay.io/amdabhad/csi-driver:prereq231222
Operaor image: quay.io/amdabhad/csi-operator:prereq231222

**Review by Jainbrt — CHANGES_REQUESTED** [review]
In case of wrong details in cluster stanza of remote cluster ( which is primary), does not let driver know that primary cluster is not usable

**Review by Jainbrt — DISMISSED** [review]

**Comment by Jainbrt** [comment]
One issue:  install CSI 2.8.0 without primaryFset value, then upgrade to 2.9.0 ,set CSO as false.

File: go.work [hunk 1]
```diff
@@ -0,0 +1,6 @@
+go 1.18
```
Comment (hemalathagajendran) [inline go.work:1]: go.work is a flexibility to test multiple module dependencies during development phase. However the recommendation is to use the tagged version of dependent module in go.mod file. go.work is recommended to place in git.ignore and not recommended for production as well. Other option is to run "go build" command with "GOWORK=OFF" and this requires changes in our dockerfile.

Comment (amdabhad) [reply go.work:1]: The current PR changes are for dev phase only and without go workspace the operator image doesn't get built locally and also on travis as operator now depends on driver module, so to merge this PR in IBM branch - the workspace file needs to be there in the PR, few of the current dependencies are pointing to my fork and I am going to adjust the dependencies after this PR is merged and I have another task already to look at these dependencies - https://github.ibm.com/ibmspectrumscale/csi-tracker/issues/81.

**Review by Jainbrt — APPROVED** [review]

**Review by ghanshyam-11 — APPROVED** [review]
looks good

**Pull request merged**